### ISSUE TITLE
Problems finding matches separated by break tag

### ISSUE THREAD

**Issue by evertsd** [padolsey/findAndReplaceDOMText#66]
Am trying to detect specific words in an external text area, and highlight them as they're found.
In said text area, words on separate lines are only separated by a `<br />` tag

For example, when `far` looks at the previous two lines, it's essentially running my regex on
```
as they're found.In said text area
```

Normally I'd detect a word break `\b` and call it a day, but as an example, if I'm trying to detect `google` I don't want it flagging on http://google.com

Perhaps I should cut my losses and not worry about how often similar use cases would occur, but figured it was worth checking if there's a simple solution I'm overlooking

### Example
https://jsfiddle.net/devdost/zoyqptzd/

*edits*
- Moved example into a JS fiddle

**Comment by padolsey** [comment]
Unfortunately, I don't think there's a general solution here, but I do that it might be easier, in this case, to write a regex that defines what you *don't* want as opposed to what you *do* want. For example, the following regex specifically avoids cases where the 'google' is followed by a period and then a word character:

```js
/google(?![.]\w)/gi
```

It [seems to work](https://jsfiddle.net/zoyqptzd/1/) with your example text.

If you want to write a more diverse set of rules for what matches to avoid, and you're finding it tricky to fit all these rules in a single regex, you could use a bit of a hacky approach to define your own filtering function ([like this](https://github.com/padolsey/findAndReplaceDOMText/issues/64#issuecomment-324686191)).

**Comment by evertsd** [comment]
Hmm, ok. Unfortunately in [my case](https://jsfiddle.net/devdost/zoyqptzd/2/) it seems that's trading one set of problems for another (sorry for not supplying a better representation of the use cases I'll encounter in the first place)

Don't take too long looking at it though, because I've found that my understanding of the markup I have to work with was also flawed. By doing a far on each immediate child, I can detect the end & beginning of lines more accurately

[Solution](https://jsfiddle.net/devdost/zoyqptzd/3/)

Closing, thanks for the advice!
